extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/DeephavenCompressorAdapterFactory.java

```diff
@@ -161,6 +167,54 @@ public void close() {
         }
     }
 
+    /**
+     * This is the default adapter for LZ4 data. It attempts to decompress with LZ4, and falls back to LZ4_RAW on
+     * failure. The fallback mechanism is particularly useful for decompressing parquet files that are compressed with
+     * LZ4_RAW but tagged as LZ4 in the metadata.
+     */
+    private static class LZ4WithLZ4RawBackupCompressorAdapter extends CodecWrappingCompressorAdapter {
+        private CompressorAdapter lz4RawAdapter = null; // Lazily initialized
+
+        private LZ4WithLZ4RawBackupCompressorAdapter(CompressionCodec compressionCodec,
+                CompressionCodecName compressionCodecName) {
+            super(compressionCodec, compressionCodecName);
+        }
+
+        @Override
+        public BytesInput decompress(final InputStream inputStream, final int compressedSize,
+                final int uncompressedSize) throws IOException {
+            // Buffer input data in case we need to retry with LZ4_RAW.
+            final BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream, compressedSize);
+            bufferedInputStream.mark(compressedSize);
+            try {
+                return super.decompress(bufferedInputStream, compressedSize, uncompressedSize);
+            } catch (IOException e) {
+                super.reset();
+                bufferedInputStream.reset();
+                if (lz4RawAdapter == null) {
+                    lz4RawAdapter = DeephavenCompressorAdapterFactory.getInstance().getByName("LZ4_RAW");
+                }
+                return lz4RawAdapter.decompress(bufferedInputStream, compressedSize, uncompressedSize);
+            }
+        }
+
+        @Override
+        public void reset() {
+            super.reset();
+            if (lz4RawAdapter != null) {
```

Comment:
to confirm, for both this and close(), we don't want to null out the lz4raw codec?

Reply:
The docs for reset: `Reset the internal state of this {@link CompressorAdapter} so more rows can be read or written.` 
So doesn't make sense to deallocate lz4raw.

For close, I can null out the codec

Reply:
I think no need to close it out if the instance cant be used again anyway - it looks like we get close() by way of SafeCloseable, so no one should attempt to use it after close().

Reply:
Cool